src/variationallaplace/optimization.jl

```diff
@@ -380,20 +380,21 @@ function _run_spDCM_iteration!(Q, Πθ_pr, Πλ_pr, μθ_pr, μλ_pr, nr, np, ny
 
     ϵ = reshape(y - f_μθ, ny)                   # error
     J = - dfdp   # Jacobian, unclear why we have a minus sign. Helmut: comes from deriving a Gaussian. 
-
     ## M-step: Fisher scoring scheme to find h = max{F(p,h)} // comment from MATLAB code
-    P = [spzeros(eltype(J), size(Q[i])) for i = 1:nh]
-    PΣ = [spzeros(eltype(J), size(Q[i])) for i = 1:nh]
-    JPJ = [spzeros(real(eltype(J)), size(J)) for i = 1:nh]
-    dFdλ = spzeros(eltype(J), nh)
-    dFdλλ = spzeros(real(eltype(J)), nh, nh)
+    eltypeJ = ComplexF64
+    P = [spzeros(eltypeJ, size(Q[i])) for i = 1:nh]
+    PΣ = [spzeros(eltypeJ, size(Q[i])) for i = 1:nh]
+    JPJ = [spzeros(real(eltypeJ), size(J)) for i = 1:nh]
+    dFdλ = spzeros(eltypeJ, nh)
+    dFdλλ = spzeros(real(eltypeJ), nh, nh)
```

Comment:
does not speed up. Seems though to increase the number of allocations from 348396 to 350301.